app/controllers/cookie_preferences_controller.rb

```diff
@@ -0,0 +1,40 @@
+class CookiePreferencesController < ApplicationController
+  def show
+    redirect_to edit_cookie_preference_path
+  end
+
+  def edit; end
+
+  def update
+    cookie_preferences.assign_attributes preference_attributes
+
+    if cookie_preferences.valid?
+      persist cookie_preferences
+      remove_rejected_cookies cookie_preferences
+      flash[:cookies] = 'updated'
+      redirect_to request.referer.presence || edit_cookie_preference_path
```

Comment:
When testing I was able to get myself stuck on the edit screen with no clear way to get back to the landing page/dashboard; the referrer in this context will always be the 'edit' path